core-bundle/src/Resources/contao/models/OptInModel.php

```diff
@@ -228,4 +228,23 @@ public function setRelatedRecords(array $arrRelated)
 			}
 		}
 	}
+
+	/**
+	 * Delete opt-in tokens by their table and ID
+	 *
+	 * @param string $strTable
+	 * @param int $intId
+	 *
+	 * @return integer The number of affected rows
+	 */
+	public static function deleteByTableAndId($strTable, $intId)
+	{
+		$t = static::$strTable;
+
+		return Database::getInstance()
+			->prepare("DELETE FROM $t WHERE $t.id = ? $t.confirmedOn = 0 AND $t.id IN (SELECT pid FROM tl_opt_in_related WHERE relTable=?)")
```

Comment:
```suggestion
			->prepare("DELETE FROM $t WHERE $t.id = ? AND $t.confirmedOn = 0 AND $t.id IN (SELECT pid FROM tl_opt_in_related WHERE relTable=?)")
```

Reply:
This still looks wrong though. I don't think we should query for ID but for relId in the subtable?

Reply:
I do query the subtable: 

`$t.id IN (SELECT pid FROM tl_opt_in_related WHERE relTable=?)")`

Reply:
yes but the member ID is in `tl_opt_in_related.relId` and not in `tl_opt_in.id`

Reply:
Like in f5f4b680d2?